sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -63,6 +65,20 @@ public bool IsLoggingEnabled
         }
     }
 
+    /// <summary>
+    /// TODO.
+    /// </summary>
+    public ILoggerFactory? LoggerFactory
+    {
```

Comment:
Opted to do ILoggerFactory over ILogger b/c:
> If you're designing a public API that must remain stable over time, keep in mind that you might desire to refactor your internal implementation in the future. Even if a class doesn't create any internal helper types initially, that might change as the code evolves. Using ILoggerFactory accommodates creating new ILogger<TCategoryName> objects for any new classes without changing the public API.

https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-library-authors#when-to-use-the-iloggerfactory-interface